docs/migration_guides/opflow_migration.rst

```diff
@@ -0,0 +1,1156 @@
+=======================
+Opflow Migration Guide
+=======================
+
+*Jump to* `TL;DR`_.
+
+Background
+----------
+
+The :mod:`~qiskit.opflow` module was originally introduced as a layer between circuits and algorithms, a series of building blocks
+for quantum algorithms research and development.
+
+The recent release of the :mod:`~qiskit.primitives` introduced a new interface for interacting with backends. Now, instead of
+preparing a circuit to execute with a ``backend.run()`` type of method, the algorithms can leverage the :class:`~Sampler` and
+:class:`~Estimator` primitives, send parametrized circuits and observables, and directly receive quasi-probability distributions or
+expectation values (respectively). This workflow simplifies considerably the pre-processing and post-processing steps
+that previously relied on opflow; encouraging us to move away from opflow, and find new paths of developing algorithms based on
+the :mod:`~qiskit.primitives` interface and the :mod:`~qiskit.quantum_info` module.
+
+This guide traverses the opflow submodules and provides either a direct alternative
+(i.e. using :mod:`~qiskit.quantum_info`), or an explanation of how to replace their functionality in algorithms.
+
+TL;DR
```

Comment:
I think the TL;DR section should be the first section, above the Background section